app/src/androidTest/java/ch/epfl/sweng/swenggolf/SavePartialOfferTest.java

```diff
@@ -117,12 +116,16 @@ public void dataIsDeletedIfWeCreateTheOffer() {
         ValueListener<Offer.Builder> listener = new ValueListener<Offer.Builder>() {
             @Override
             public void onDataChange(Offer.Builder value) {
-                assertNull(value);
+                fail();
             }
 
             @Override
             public void onCancelled(DbError error) {
-                fail();
+                if (error == DbError.DATA_DOES_NOT_EXIST){
```

Comment:
Maybe you could do a more concise check? Something like `assertThat(error, is(DbError.DATA_DOES_NOT_EXIST))`? (you actually use it in multiple places)

Reply:
You are absolutely right, I corrected it and the 2 other locations where I did a similar thing.